cpp/src/slice2swift/SwiftUtil.cpp

```diff
@@ -1803,7 +1801,7 @@ SwiftGenerator::MetadataVisitor::visitModuleStart(const ModulePtr& p)
             ostringstream os;
             os << "invalid module mapping:\n Slice module `" << p->scoped() << "' should be map to Swift module `"
                << current->second << "'" << endl;
-            dc->error(p->file(), p->line(), os.str());
+            unit->error(p->file(), p->line(), os.str());
```

Comment:
The error message above is not grammatically correct.

Also, for my understanding, what's the difference between p->file()/p->line() and the overload without them?

Reply:
If you don't provide a `file` and `line` parameter, the parser/`Unit` will use `currentFile()` and `currentLine()` for them. This is useful _while_ parsing is occurring, since these are almost always what we want to pass.

Once we hit the visitors though, parsing has already finished, so `currentFile` and `currentLine` will be the _last_ file and line the parser saw, which isn't what we want.

----

Also I fixed the error message.